File: Lato Client/src/Person.ts
export abstract class Person {
    id: string;
    firstName: string;
    lastName: string;
    dateOfBirth: Date;
    domain: string;

    constructor(id: string, FirstName: string, LastName: string, dateOfBirth: Date, domain: string) {
        this.id = id;
        this.firstName = FirstName;
        this.lastName = LastName;
        this.dateOfBirth = dateOfBirth;
        this.domain = domain;
    }
    get email(): string {
        return `${this.firstName[0].toLowerCase()}${this.lastName.toLowerCase()}@${this.domain.toLowerCase()}.com`
    }

    abstract getDetails(): string;

}

File: Lato Server/Lato Server/Person.cs
namespace Lato_Server
{
    public class Person
    {
        public Guid Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string Domain { get; set; }

        public string Email { get; set; }

        public Person() { }
    }
}

//    public class Student : Person
//    {
//        public Student() { }

//        public override string getPersonDetails()
//        {
//            return $"Studente: {Id} | {FirstName} {LastName} | Età: | Email: {Email}";
//        }

//    }

//    public class Teacher : Person
//    {
//        public Teacher() { }

//        public override string getPersonDetails()
//        {
//            return $"Docente: {Id} | {FirstName} {LastName} | Età: | Email: {Email}";
//        }
//    }
//}

